lib/src/frontend/future_matchers.dart

```diff
@@ -80,3 +80,54 @@ class _Completes extends AsyncMatcher {
     return description;
   }
 }
+
+/// Matches a [Future] that does not complete.
+///
+/// Note that this creates an asynchronous expectation. The call to
+/// `expect()` that includes this will return immediately and execution will
+/// continue. Later, after 20  passes through the event queue,
+/// the matcher will complete.
+///
+/// This returns an [AsyncMatcher], so [expect] won't complete until
+/// [timesToPump] passes through the event queue.
```

Comment:
There are other ways to keep the test open. The underlying mechanism is [`Invoker.add/removeOutstandingCallback()`](https://github.com/dart-lang/test/blob/master/lib/src/backend/invoker.dart#L150-L170), but in this case it might be simpler to just run `expect(_pumpEventQueue(), completes)` within the matcher body.